.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,304 @@
+ - Use `camelCase` for naming custom functions and variables.
+ - Use `PascalCase` for defining custom types.
+### Formatting rules
+Since MeTTa lacks a flexible formatter, formatting is done as you write code. Some basic formatting conventions we use for MeTTa include:
+- **Function calls**:
+    - If the parameters are short or few, keep the function call on a single line.
+    - If the parameters are long or numerous, the function name should stick with the 
+      left-most bracket, and the parameters should be placed on the next lines with 
+      increased indentation. Example:
+	```scheme
+	;; Example function with short parameter
+
+	;; Example function with long parameters
+    (foo 
+	    (and (== $x $y) (< $x 100))
+		(X is valid parameter)
+		(X isn't valid parameter)
+    )
+
+	;; Example function with many parameters
+	(let* 
+		(
+			($x (someFunction1 ...))
+			($y (someFunction2 ...))
+			($z (someFunction3 ...))
+		) 
+		($x $y $z)
+	)
+	```
+	- Whenever calling the `if` function, the function call should be a one line if the `$then` and `$else` cases are simple. But if either one of them is complex, the function name and the condition should be on the same line as the opening bracket and  the rest of the parameters should be on the next line with the same level of indentation as the condition. Example:
+	```scheme
+	;; Calling if with simple $then and $else condtions
+	(if (== $x 1) True False)
+	
+	;; Calling if with complex $then and $else condtions
+	(if (== $x 1)
+		True
+		(let* 
+			(
+				($x' (> $x 0))
+				($result (< $x 10))
+			)
+			(and $x' $result)
+		)
+	)
+	```
+
+  - **Multi-line functions**: The closing bracket of a function should match the indentation level of its opening bracket.
+  - **Indentation**: Increase indentation level for nested parameters or function calls for better readability.
+### Function Definitions
+- When creating multiple definition of a function, don't add empty lines between the definitions 
+  unless the definitions are complex. For complex multi-line definitions, add a single line 
+  between them for clarity.
+- No empty lines should exist between a function and its type signature or its documentation.
+- When writing complex function definition, the name of the function and it's parameters should be on the same line and the rest of the function definition should start on a new line with the same level of indentation as the function name.
+Example:
+```scheme
+;; Example function that does nothing with simple definitions.
+(: foo (-> Atom Atom))
+(= (foo $x) ($x))
+(= (foo $x $x) ($x))
+(= (foo $x $y) ($x $y))
+(= (foo $x $y $z) ($x $y $z))
+
+;; Example function with complex definitions. 
+(: foo' (-> Atom Atom))
+(= (foo $x) 
+   (if (...)
+	   (...)
+	   (...)
+   )
+)
+
+(= (foo $x $y)
+   (let $bar (if (== $x $y) (...) (...)))
+)
+```
+
+### Commenting guidelines
+- Always use double semicolons (`;;`) for comments instead of a single semicolon (`;`) to maintain consistency.
```

Comment:
@Yagth its great to have common comment to maintain consistency but for example in my case my code editor make single semicolon by default so i am afraid it creates hassle making every comment double semicolon. you can suggest me your formatter extension though.

Reply:
I am using vim and its easy to set it up there. However I think it is doable for smaller comments at least.